cli/iconBuilder.js

```diff
@@ -0,0 +1,167 @@
+			});
+		});
+	});
+}
+
+async function getSvgs() {
+	return new Promise(async(resolve, reject) => {
+		fs.readdir(imagePath, async(err, files) => {
+			if (err) return reject(err);
+			let svgs = [];
+			for (let i = 0; i < files.length; i++) {
+				const newSvgs = await getSvgsInDir(files[i]);
+				svgs = svgs.concat(newSvgs);
+				if (i === files.length - 1) {
+					resolve(svgs);
+				}
+			}
+		});
+	});
+}
+
+function clearDirectoryEntry(entryPath) {
+	return new Promise(async(resolve, reject) => {
+		fs.stat(entryPath, async(err, stat) => {
+			if (err) return reject(err);
+			if (stat.isDirectory()) {
+				await clearDirectory(entryPath);
+				fs.rmdir(entryPath, err => {
+					if (err) reject(err);
+					resolve();
+				});
+			} else {
+				fs.unlink(entryPath, err => {
+					if (err) reject(err);
+					resolve();
+				});
+			}
+		});
+	});
+}
+
+async function clearDirectory(dirPath) {
+	return new Promise(async(resolve, reject) => {
+		fs.readdir(dirPath, async(err, files) => {
+			if (err) return reject(err);
+			const promises = [];
+			for (const file of files) {
+				const newPath = path.join(dirPath, file);
+				promises.push(clearDirectoryEntry(newPath));
+			}
+			resolve(Promise.all(promises));
+		});
+	});
+}
+
+async function createCategory(category) {
+	return new Promise((resolve) => {
+		const categoryPath = path.join(outputPath, category);
+		fs.mkdir(categoryPath, () => {
+			resolve();
+		});
+	});
+}
+
+async function createSvg(entry) {
+	return new Promise(async(resolve, reject) => {
+
+		await createCategory(entry.category);
+
+		const sourcePath = path.join(imagePath, entry.category, `${entry.name}.svg`);
+		const destPath = path.join(outputPath, entry.category, `${entry.name}.js`);
+
+		fs.readFile(sourcePath, (err, data) => {
+			if (err) return reject(err);
+			/*const output = 'import { html } from "../../../../lit-element/lit-element.js";\n' +
+				'export const val = html`' + data + '`;';*/
+			const output = 'export const val = `' + data + '`;';
+			fs.writeFile(destPath, output, err => {
+				if (err) return reject(err);
+				resolve();
+			});
+		});
+
+	});
+}
+
+async function createSvgs(svgs) {
+	return Promise.all(
+		svgs.map((entry) => {
+			return createSvg(entry);
+		})
+	);
+}
+
+async function createLoader(svgs) {
+	return new Promise((resolve, reject) => {
+
+		let template = '// auto-generated\n' +
+			'export async function loadSvg(icon) {\n' +
+			'\tlet svg;\n' +
+			'\tswitch (icon) {\n';
+		svgs.forEach((svg) => {
+			template += `\t\tcase 'd2l-${svg.category}:${svg.name}':
+			svg = await import('./${svg.category}/${svg.name}.js');
+			break;\n`;
+		});
+		template += '\t}\n' +
+			'\treturn svg ? svg.val : undefined;\n' +
+			'}';
+
+		const filePath = path.join(outputPath, 'presetIconLoader.js');
+		fs.writeFile(filePath, template, err => {
+			if (err) return reject(err);
+			resolve();
+		});
+
+	});
+}
+
+async function generate() {
+
+	console.log('Generating icons...');
```

Comment:
If you want to avoid the `no-console` linting exception, I think it would be possible to switch these logs to `process.stdout.write(...)`.  If you're interested, [chalk](https://www.npmjs.com/package/chalk) is what I used to get fancy colors... though I think Travis generally ignored the colors so I only see them locally.

Reply:
Cool, will switch it over -- thanks!

Reply:
`stdout` doesn't have the handy `.group()`... I've used chalk before but it also doesn't have grouping. Hmm...

Reply:
I see, the linting exception is just a minor annoyance. --shrug--